StakingERC721/contracts/StakingERC721.sol

```diff
@@ -0,0 +1,371 @@
+    stakingToken = IERC721(stakingToken_);
+    rewardsDuration = duration;
+  }
+
+  function connectToOtherContracts(
+    address[] memory otherContracts
+  ) external override onlyOwner {}
+
+  /**
+   * @notice Pause or unpause contract
+   * @dev Callable by owner
+   * @param newPaused Flag to new paused state
+   */
+  function setPaused(bool newPaused) external onlyOwner {
+    if (newPaused) {
+      _pause();
+    } else {
+      _unpause();
+    }
+  }
+
+  /**
+   * @notice Set rewards duration, only available to set after finish
+   * of previous rewards period.
+   * @dev Callable by owner
+   * @param duration New rewards duration
+   */
+  function setRewardsDuration(uint256 duration) external onlyOwner {
+    if (block.timestamp < periodFinish) {
+      revert WaitToFinish();
+    }
+    rewardsDuration = duration;
+    emit RewardsDurationUpdated(duration);
+  }
+
+  /**
+   * @notice Stakes user's NFTs
+   * @dev Callable only when unpaused
+   * @param tokenIds The tokenIds of the NFTs which will be staked
+   */
+  function stake(
+    uint256[] memory tokenIds
+  ) public nonReentrant whenNotPaused updateReward(msg.sender) {
+    if (tokenIds.length == 0) {
+      revert NoTokenIds();
+    }
+
+    uint256 amount;
+    uint256 length = tokenIds.length;
+    for (uint256 i = 0; i < length; ++i) {
+      // Transfer user's NFTs to the staking contract
+      stakingToken.safeTransferFrom(msg.sender, address(this), tokenIds[i]);
+      // Increment the amount which will be staked
+      ++amount;
+      // Save who is the staker/depositor of the token
+      stakedAssets[tokenIds[i]] = msg.sender;
+    }
+    _stake(amount);
+    emit Staked(msg.sender, amount, tokenIds);
+  }
+
+  /**
+   * @notice Unstake staked user's NFTs
+   * @dev Callable only when unpaused
+   * @param tokenIds The tokenIds of the NFTs which will be withdrawn
+   */
+  function unstake(
+    uint256[] memory tokenIds
+  ) public nonReentrant whenNotPaused updateReward(msg.sender) {
+    if (tokenIds.length == 0) {
+      revert NoTokenIds();
+    }
+
+    uint256 amount;
+    uint256 length = tokenIds.length;
+    for (uint256 i = 0; i < length; ++i) {
+      // Check if the user who withdraws is the owner
+      if (stakedAssets[tokenIds[i]] != msg.sender) {
+        revert NotTokenOwner();
+      }
+      // Transfer NFTs back to the owner
+      stakingToken.safeTransferFrom(address(this), msg.sender, tokenIds[i]);
+      // Increment the amount which will be withdrawn
+      ++amount;
+      // Cleanup stakedAssets for the current tokenId
+      stakedAssets[tokenIds[i]] = address(0);
+    }
+    _unstake(amount);
+
+    emit Unstaked(msg.sender, amount, tokenIds);
+  }
+
+  /**
+   * @notice Claim rewards tokens, callable only when unpaused.
+   * @dev Callable only when unpaused
+   */
+  function claim() public nonReentrant whenNotPaused updateReward(msg.sender) {
+    uint256 reward = rewards[msg.sender];
+    if (reward > 0) {
```

Comment:
Please add require statement.
The client waste gas fee even though there is not to claim.